File: ios/Helpers/ItemView.swift
import Foundation
import heresdk

@objc
public protocol ItemFeatureEvents {
  func updateFeature()
  func removeFeature()
}

@objc(ItemView)
public class ItemView: UIView {

  weak public var parentMap: MapView?

  public func assignToMap(map: MapView) {
    parentMap = map
  }
  
  public func unassignMap() {
    parentMap = nil
  }
}

File: android/src/main/java/com/hereexplore/ItemView.kt
package com.hereexplore

import android.content.Context
import com.facebook.react.views.view.ReactViewGroup

abstract class ItemView(context: Context?) : ReactViewGroup(context) {

  protected var parentMap: MapsView? = null

  abstract fun updateFeature()

  abstract fun removeFeature()

  fun assignToMap(map: MapsView) {
    parentMap = map
  }

  fun unassignMap() {
    parentMap = null
  }
}
